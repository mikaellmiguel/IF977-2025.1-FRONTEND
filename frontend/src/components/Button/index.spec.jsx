import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Button } from "./index";

describe("Button", () => {
  it("renderiza o texto corretamente", () => {
    const { getByText } = render(<Button>Salvar</Button>);
    expect(getByText("Salvar")).toBeInTheDocument();
  });

  it("passa props corretamente para o botÃ£o", () => {
    const { getByRole } = render(<Button type="submit" data-testid="btn" />);
    const btn = getByRole("button");
    expect(btn).toHaveAttribute("type", "submit");
    expect(btn).toHaveAttribute("data-testid", "btn");
  });

  it("executa onClick quando clicado", () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button onClick={handleClick}>Clique</Button>);
    fireEvent.click(getByText("Clique"));
    expect(handleClick).toHaveBeenCalled();
  });

  it("aplica estilos de hover e active (snapshot)", () => {
    const { container } = render(<Button>Hover</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
