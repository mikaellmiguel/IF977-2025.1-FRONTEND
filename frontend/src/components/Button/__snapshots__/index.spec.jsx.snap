// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button aplica estilos de hover e active (snapshot) 1`] = `
<button
  class="sc-bRKDuR jDSreh"
>
  Hover
</button>
`;
